devito/passes/clusters/misc.py

```diff
@@ -30,8 +30,8 @@ def callback(self, clusters, prefix):
         lifted = []
         processed = []
         for n, c in enumerate(clusters):
-            # Increments prevent lifting
-            if c.has_increments:
+            # Increments and implicit dimensions prevent lifting
+            if c.has_increments or c.has_implicit_dims:
```

Comment:
as Rhodri has pointed out, this smells like a hacky workaround

Reply:
This has been moved to a configuration option.